### ISSUE TITLE
Object cannot use `with` to reach the scope in itself

### ISSUE THREAD

**Issue by shichen85** [YoYoGames/GameMaker-Bugs#6765]
### Description
If an object called Object1 execute the following code:
```gml
	with Object1 {
		show_debug_message("Can reach");
		return;
	}
	throw("Cannot reach");
```
It will skip the return statement and run the throw statement. This behavior is different from the 600 runtimes.

### Steps To Reproduce
1. Run the project in VM windows, see that error is thrown.
2. Comment out Object1's creation event and see that Object2's creation event runs without issue.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.800.0.593 Runtime v2024.800.0.617

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22000.0

### Which platform(s) are you seeing the problem on?
iOS, Windows

[1003a073-59e1-4f05-8f9a-184ecbad10df](https://api.gamemaker.io/api/github/downloads/1003a073-59e1-4f05-8f9a-184ecbad10df)

**Comment by shichen85** [comment]
Specifically, the code has to execute in the Creation event for this issue to occur.

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by mgeddesGM** [comment]
verified as of Beta IDE v2024.800.0.593  Runtime v2024.800.0.618
